ambry-cloud/src/main/java/com/github/ambry/cloud/VcrServer.java

```diff
@@ -74,6 +75,7 @@
   private JmxReporter reporter = null;
   private ConnectionPool connectionPool = null;
   private final NotificationSystem notificationSystem;
+  private final Function<MetricRegistry, JmxReporter> reporterFactory;
```

Comment:
I like this Function usage.  It's inconsistent with the other factory interfaces we use, but it seems like a sensible shortcut.

Reply:
I was going to add a Factory instance but putting it in a place where all 3 entry points (server, frontend, vcr) could share it without adding a dependency on metrics-jmx to a package became a problem (metrics-jmx is where the JmxReporter class is defined)